credentials/credentials.go

```diff
@@ -193,6 +194,33 @@ func RequestInfoFromContext(ctx context.Context) (ri RequestInfo, ok bool) {
 	return
 }
 
+// ClientHandshakeInfo holds data to be passed to ClientHandshake. This makes
+// it possible to pass arbitrary data to the handshaker from gRPC, resolver,
+// balancer etc. Individual credential implementations control the actual
+// format of the data that they are willing to receive.
+//
+// This API is experimental.
+type ClientHandshakeInfo struct {
+	// Attr is a generic key/value store.
+	Attr *attributes.Attributes
+}
+
+// clientHandshakeInfoKey is a struct used as the key to store
+// ClientHandshakeInfo in a context.
+type clientHandshakeInfoKey struct{}
+
+// WithClientHandshakeInfo returns a copy of parent with chi stored as a value.
```

Comment:
```
//
// This API is EXPERIMENTAL.
```

And below please.

Actually, this one is for internal use only, correct?  Let's follow the same pattern as `RequestInfo` if so:

https://github.com/grpc/grpc-go/blob/master/internal/internal.go#L42:1